### ISSUE TITLE
Unable to clone xgboost submodule 'cub'

### ISSUE THREAD

**Issue by Yunni** [dmlc/xgboost#2594]
## Environment info
Operating System:
- System Version: macOS 10.12.6
- Kernel Version: Darwin 16.7.0

Compiler: Does not matter

Package used (python/R/jvm/C++):
- git version 2.13.2

`xgboost` version used:
- master branch

## Steps to reproduce

```
git clone --recursive https://github.com/dmlc/xgboost
Cloning into 'xgboost'...
remote: Counting objects: 21177, done.
remote: Compressing objects: 100% (21/21), done.
remote: Total 21177 (delta 2), reused 2 (delta 0), pack-reused 21156
Receiving objects: 100% (21177/21177), 7.41 MiB | 1.61 MiB/s, done.
Resolving deltas: 100% (12540/12540), done.
Submodule 'cub' (https://github.com/NVlabs/cub) registered for path 'cub'
Submodule 'dmlc-core' (https://github.com/dmlc/dmlc-core) registered for path 'dmlc-core'
Submodule 'nccl' (https://github.com/dmlc/nccl) registered for path 'nccl'
Submodule 'rabit' (https://github.com/dmlc/rabit) registered for path 'rabit'
Cloning into '/Users/yunn/xgboost/cub'...
remote: Counting objects: 180, done.
remote: Compressing objects: 100% (162/162), done.
remote: Total 180 (delta 53), reused 52 (delta 13), pack-reused 0
Receiving objects: 100% (180/180), 489.53 KiB | 430.00 KiB/s, done.
Resolving deltas: 100% (53/53), done.
Cloning into '/Users/yunn/xgboost/dmlc-core'...
remote: Counting objects: 4320, done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 4320 (delta 1), reused 3 (delta 1), pack-reused 4314
Receiving objects: 100% (4320/4320), 975.52 KiB | 518.00 KiB/s, done.
Resolving deltas: 100% (2559/2559), done.
Cloning into '/Users/yunn/xgboost/nccl'...
remote: Counting objects: 652, done.
remote: Total 652 (delta 0), reused 0 (delta 0), pack-reused 652
Receiving objects: 100% (652/652), 1.38 MiB | 594.00 KiB/s, done.
Resolving deltas: 100% (411/411), done.
Cloning into '/Users/yunn/xgboost/rabit'...
remote: Counting objects: 3142, done.
remote: Total 3142 (delta 0), reused 0 (delta 0), pack-reused 3142
Receiving objects: 100% (3142/3142), 900.20 KiB | 538.00 KiB/s, done.
Resolving deltas: 100% (2039/2039), done.
error: Server does not allow request for unadvertised object f3937a96fdec78a73446aaaa114c112ff31f5503
Fetched in submodule path 'cub', but it did not contain f3937a96fdec78a73446aaaa114c112ff31f5503. Direct fetching of that commit failed.
```

**Comment by daizhongxiang** [comment]
I'm doing it in Windows 10, the same erorr happened when I tried to clone:

error: Server does not allow request for unadvertised object f3937a96fdec78a73446aaaa114c112ff31f5503
Fetched in submodule path 'cub', but it did not contain f3937a96fdec78a73446aaaa114c112ff31f5503. Direct fetching of that commit failed.

**Comment by fantopop** [comment]
I have got the same error today (macOS 10.12.6, git 2.14.1) performing command from installation page:
`git clone --recursive https://github.com/dmlc/xgboost  `
...
`error: Server does not allow request for unadvertised object f3937a96fdec78a73446aaaa114c112ff31f5503
Fetched in submodule path 'cub', but it did not contain f3937a96fdec78a73446aaaa114c112ff31f5503. Direct fetching of that commit failed.`

**Comment by RAMitchell** [comment]
Should be fixed in #2591

**Comment by Andy1314Chen** [comment]
I have got the same error !  （windows7）

>> error: Server does not allow request for unadvertised object f3937a96fdec78a7344                                 6aaaa114c112ff31f5503
Fetched in submodule path 'cub', but it did not contain f3937a96fdec78a73446aaaa                                 114c112ff31f5503. Direct fetching of that commit failed.
